app/controllers/tweets_controller.rb

```diff
@@ -7,12 +7,42 @@ def index
       config.access_token        = ENV["TWITTER_ACCESS_TOKEN"]
       config.access_token_secret = ENV["TWITTER_ACCESS_SECRET"]
     end
-    query = params[:search]
-    @tweets = client.search("##{query}").take(10)
 
-    if current_user
-      Search.create(query: params[:search], trend_or_location: true, user: current_user)
+    if params[:search]
+      query = params[:search]
+      @tweets = client.search("##{query}").take(25)
+
+      if current_user
+        Search.create(query: params[:search], trend_or_location: "Trend", user: current_user)
+      end
+
+    elsif params[:local]
+      ip_address = request.remote_ip unless Rails.env.test? || Rails.env.development?
+        ip_address = "50.241.127.209" if Rails.env.test? || Rails.env.development?
+      location = GeoIP.new('GeoLiteCity.dat').city(ip_address)
+      q = "geocode:#{location.latitude},#{location.longitude},1mi"
+      @tweets = client.search(q).take(25)
+
+      if current_user
+        Search.create(query: location.real_region_name, trend_or_location: "Location", user: current_user)
+      end
+
+    elsif params[:location]
+      location = params[:location]
+      query = HTTParty.get("https://maps.googleapis.com/maps/api/geocode/json?address=#{location}&key=#{ENV["GOOGLE_GEOCODING"]}")
```

Comment:
Line is too long. [130/80]<br>Prefer single-quoted strings inside interpolations.